public/js/triage/triage.js

```diff
@@ -673,35 +675,23 @@ $(document).ready(function () {
         checkIfDuplicatePatient();
     });
 
-    // function checkIfDuplicatePatient() {
-    //     var patientInfo = triageFields.patientInformation;
-    //     var query = patientInfo.firstName.val() + " " + patientInfo.lastName.val();
-    //     var url = "/search/check/" + query;
-    //     var patientId = $("#patientId").val();
-    //
-    //     $.getJSON(url, function (result) {
-    //         if (result === true) {
-    //             if(!(patientId > 0)) {
-    //                 if (confirm("A patient with this name already exists in the database. Would you like to view the matching patient information?")) {
-    //                     var duplicatePatientUrl = "/history/patient/" + patientInfo.firstName.val() + "-" + patientInfo.lastName.val();
-    //                     window.location.replace(duplicatePatientUrl);
-    //                 }
-    //             }
-    //         }
-    //     })
-    // };
-
     function checkIfDuplicatePatientMatch() {
         var patientInfo = triageFields.patientInformation;
         var first = patientInfo.firstName.val();
         var last = patientInfo.lastName.val();
-        var url = "/search/findMatch/" + first + "/" + last;
+        var phone = patientInfo.phoneNumber.val();
+        var addr = patientInfo.address.val();
+        var age = patientInfo.years.val();
```

Comment:
we might need to fix the age later on due to the different age inputs and the way its calculated

Reply:
I changed the search function to not compared the age number value but instead compare the date string (which is what is being stored in the db). I used the same converter function that already existed to get the date from age, then converted the date to type long to be passed over the router. Once I get the date as a long in the java code I convert it back to a date object then a string to be used in the ranking query.